client/src/components/ViewPostDialog.tsx

```diff
@@ -0,0 +1,262 @@
+import AppBar from "@mui/material/AppBar";
+import IconButton from "@mui/material/IconButton";
+import Typography from "@mui/material/Typography";
+import ArrowBack from "@mui/icons-material/ArrowBack";
+import Slide from "@mui/material/Slide";
+import { TransitionProps } from "@mui/material/transitions";
+import MoreVert from "@mui/icons-material/MoreVert";
+import Menu from "@mui/material/Menu";
+import MenuItem from "@mui/material/MenuItem";
+import Stack from "@mui/material/Stack";
+import Box from "@mui/material/Box";
+import LinearProgress from "@mui/material/LinearProgress";
+import TextField from "@mui/material/TextField";
+import ThumbUpIcon from "@mui/icons-material/ThumbUp";
+import ThumbUpOffAltIcon from "@mui/icons-material/ThumbUpOffAlt";
+import GenerateTags from "./Tags";
+
+const Transition = React.forwardRef((
+  props: TransitionProps & {
+    children: React.ReactElement;
+  },
+  ref: React.Ref<unknown>
+) => {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+/* Post settings, choosing between deleting, editing or reporting a post. The delete
+  and edit options are only shown if the user is authorized. */
+function MoreOptions() {
+  const [isOpen, openOptions] = React.useState(false);
+
+  const handleClick = () => {
+    openOptions(true);
+  };
+  const handleClose = () => {
+    openOptions(false);
+  };
+
+  return (
+    <>
+      <IconButton
+        id="post-settings"
+        data-testid="test-post-settings"
+        color="inherit"
+        aria-controls="settings-menu"
+        aria-haspopup="true"
+        aria-expanded={isOpen}
+        onClick={handleClick}
+      >
+        <MoreVert />
+      </IconButton>
+      <Menu
+        id="post-settings-menu"
+        data-testid="test-post-settings-menu"
+        anchorEl={document.getElementById("post-settings")}
+        open={isOpen}
+        onClose={handleClose}
+        MenuListProps={{
+          "aria-labelledby": "post-settings",
+        }}
+      >
+        <MenuItem onClick={handleClose}>Edit</MenuItem>
+        <MenuItem onClick={handleClose}>Delete</MenuItem>
+        <MenuItem onClick={handleClose}>Report</MenuItem>
+      </Menu>
+    </>
+  );
+}
+
+/* Like button. Handles liking/unliking a post */
+function LikeButton() {
+  // TODO: update/get data from db
+  const [isLiked, likePost] = React.useState(false);
+
+  const handleClick = () => {
+    likePost((prevLike) => !prevLike);
+  };
+
+  const likeButton = isLiked ? (
+    <ThumbUpIcon onClick={handleClick} fontSize="large" />
+  ) : (
+    <ThumbUpOffAltIcon onClick={handleClick} fontSize="large" />
+  );
+
+  return (
+    <Stack>
+      {likeButton}
+      <Typography sx={{ px: 2 }}>0</Typography>
+    </Stack>
+  );
+}
+
+function CapacityBar() {
+  const [capacity, setCapacity] = React.useState(0);
+  const [checkedIn, setCheckIn] = React.useState(false);
+  const maxCapacity = 10; // TODO: Get from db
+
+  const handleCheckIn = () => {
```

Comment:
Perhaps make this `toggleCheckin` and have it toggle state. You can +-1 based on the value of checkedIn. You also won't need `handleUndo`